src/coreclr/nativeaot/System.Private.CoreLib/src/System/EETypePtr.cs

```diff
@@ -347,34 +339,47 @@ internal CorElementType CorElementType
         {
             get
             {
-                Debug.Assert((int)CorElementType.ELEMENT_TYPE_BOOLEAN == (int)EETypeElementType.Boolean);
-                Debug.Assert((int)CorElementType.ELEMENT_TYPE_I1 == (int)EETypeElementType.SByte);
-                Debug.Assert((int)CorElementType.ELEMENT_TYPE_I8 == (int)EETypeElementType.Int64);
-                EETypeElementType elementType = ElementType;
-
-                if (elementType <= EETypeElementType.UInt64)
-                    return (CorElementType)elementType;
-                else if (elementType == EETypeElementType.Single)
-                    return CorElementType.ELEMENT_TYPE_R4;
-                else if (elementType == EETypeElementType.Double)
-                    return CorElementType.ELEMENT_TYPE_R8;
-                else if (elementType == EETypeElementType.IntPtr)
-                    return CorElementType.ELEMENT_TYPE_I;
-                else if (elementType == EETypeElementType.UIntPtr)
-                    return CorElementType.ELEMENT_TYPE_U;
-                else if (IsValueType)
-                    return CorElementType.ELEMENT_TYPE_VALUETYPE;
-                else if (IsByRef)
-                    return CorElementType.ELEMENT_TYPE_BYREF;
-                else if (IsPointer)
-                    return CorElementType.ELEMENT_TYPE_PTR;
-                else if (IsSzArray)
-                    return CorElementType.ELEMENT_TYPE_SZARRAY;
-                else if (IsArray)
```

Comment:
I have noticed redundant `IsSzArray` check in the original change. I started typing suggestion for get rid of it, but then realized that this whole method is very inefficient and ended up with this. 

This property is a simple branch-free lookup with this change:
```
Test.get_CorElementType()
    L0000: mov eax, [rcx+8]
    L0003: shr eax, 0x1a
    L0006: and eax, 0x1f
    L0009: mov rdx, 0x25844e91168
    L0013: movzx eax, byte ptr [rax+rdx]
    L0017: ret
```

Reply:
Nice!